### ISSUE TITLE
connect to ES with template_name and template_file my td-agent wont start.

### ISSUE THREAD

**Issue by aonamrata** [uken/fluent-plugin-elasticsearch#382]
#### Problem
I have same problem as https://github.com/uken/fluent-plugin-elasticsearch/issues/241
When I connect to ES with template_name and template_file my td-agent wont start. But If i comment the lines they work.
...

#### Steps to replicate
I get the logs from UDP i/p and write them to ES. my index are created by `logstash_format true` so its not a single index, but gets created for each day. I want my data to get mapping from template instead of default mapping provided by ES.

My td-agent.cofig looks like https://gist.github.com/aonamrata/993889b073fb9e5cf84b511a8c6ff887#file-td-agent-conf
My template looks like https://gist.github.com/aonamrata/993889b073fb9e5cf84b511a8c6ff887#file-template3-json

Note both ES host and the template file are accessible and running. So not a n/w thing.

Error on restart:
```
-bash-4.2$ sudo /etc/init.d/td-agent restart
Restarting td-agent (via systemctl):  Job for td-agent.service failed because the control process exited with error code. See "systemctl status td-agent.service" and "journalctl -xe" for details.
                                                           [FAILED]
-bash-4.2$ sudo journalctl -xe
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/plugin.rb:164:in `configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/agent.rb:130:in `add_match'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/agent.rb:72:in `block in configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/agent.rb:64:in `each'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/agent.rb:64:in `configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/root_agent.rb:112:in `configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/engine.rb:131:in `configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/engine.rb:96:in `run_configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/supervisor.rb:770:in `run_configure'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/supervisor.rb:553:in `dry_run'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/supervisor.rb:571:in `supervise'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/supervisor.rb:476:in `run_supervisor'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/command/fluentd.rb:310:in `<top (required)>'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/bin/fluentd:8:in `<top (required)>'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/bin/fluentd:22:in `load'
Mar 16 06:26:09 dev-httpd-vr fluentd[8344]: from /opt/td-agent/embedded/bin/fluentd:22:in `<main>'
Mar 16 06:26:09 dev-httpd-vr systemd[1]: td-agent.service: control process exited, code=exited status=1
Mar 16 06:26:09 dev-httpd-vr systemd[1]: Failed to start td-agent: Fluentd based data collector for Treasure Data.
-- Subject: Unit td-agent.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit td-agent.service has failed.
--
-- The result is failed.
Mar 16 06:26:09 dev-httpd-vr systemd[1]: Unit td-agent.service entered failed state.
Mar 16 06:26:09 dev-httpd-vr systemd[1]: td-agent.service failed.
Mar 16 06:26:10 dev-httpd-vr systemd[1]: td-agent.service holdoff time over, scheduling restart.
Mar 16 06:26:10 dev-httpd-vr systemd[1]: start request repeated too quickly for td-agent.service
Mar 16 06:26:10 dev-httpd-vr systemd[1]: Failed to start td-agent: Fluentd based data collector for Treasure Data.
-- Subject: Unit td-agent.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit td-agent.service has failed.
--
-- The result is failed.
Mar 16 06:26:10 dev-httpd-vr systemd[1]: Unit td-agent.service entered failed state.
Mar 16 06:26:10 dev-httpd-vr systemd[1]: td-agent.service failed.
Mar 16 06:26:16 dev-httpd-vr sudo[8347]:  devorch : TTY=pts/2 ; PWD=/home/devorch ; USER=root ; COMMAND=/usr/bin/journalctl -xe
lines 1451-1492/1492 (END)
```

#### Expected Behavior or What you need to ask
- td-agent should start successfully and 
- when i check the newly created index on ES `curl -XGET 'localhost:9200/ows1-20190404/_mapping?pretty'` it should have the mapping as in template3.json
...

#### Using Fluentd and ES plugin versions

* centos
* td-agent-3.1.1-0.el7.x86_64 => v1.0 (td-agent3)
* td-agent gems: 
```
elasticsearch (5.0.4)
elasticsearch-api (5.0.4)
elasticsearch-transport (5.0.4)
excon (0.60.0)
faraday (0.13.1)
fluent-logger (0.7.1)
fluent-plugin-elasticsearch (2.8.1, 2.4.0)
fluent-plugin-kafka (0.6.5)
fluent-plugin-rewrite-tag-filter (2.0.1)
fluent-plugin-s3 (1.1.0)
fluent-plugin-td (1.0.0)
fluent-plugin-td-monitoring (0.2.3)
fluent-plugin-webhdfs (1.2.2)
fluentd (1.0.2)
```
* ES version (optional)
```
{
  "name" : "4c65VAg",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "m6DznFJHSNulS41sJ69Taw",
  "version" : {
    "number" : "6.2.2",
    "build_hash" : "10b1edd",
    "build_date" : "2018-02-16T19:01:30.685723Z",
    "build_snapshot" : false,
    "lucene_version" : "7.2.1",
    "minimum_wire_compatibility_version" : "5.6.0",
    "minimum_index_compatibility_version" : "5.0.0"
  },
  "tagline" : "You Know, for Search"
}
```

**Comment by aonamrata** [comment]
Updated `fluentd (1.1.1)` still the same.

**Comment by cosmo0920** [comment]
Installed elasticsearch gem v5.0.4 is compatible for ES v5.
Updating  elasticsearch gem v6.0.1 is same result?

**Comment by aonamrata** [comment]
Let me try. btw, I downgraded this plugin to v2.6.1 and that works.

**Comment by cosmo0920** [comment]
How about set `content_type application/json` in your fluentd.conf?
I guess this parameter causes this issue.

**Comment by aonamrata** [comment]
:+1: That did it. Updating all gems did not do it but `content_type` did
```
<store>
       @type elasticsearch
      host localhost
      port 9200

      include_tag_key true
      tag_key source_tag
      # flush_interval 10s
      type_name ows1
      content_type application/json

      logstash_format true
      logstash_prefix ows1
      logstash_dateformat %Y%m%d
      time_key timestamp
      time_key_exclude_timestamp true

      template_name template3
      template_file /home/devorch/template3.json
      template_overwrite true

  </store>
```
Should we add this to docs?
Thanks.

**Comment by cosmo0920** [comment]
> Should we add this to docs?

I'll revert default content_type to `application/json` from `application/x-ndjson`.
This change has been introduced in https://github.com/uken/fluent-plugin-elasticsearch/pull/367 to fix #366.

And it is not similar issue of #241.
Root cause of problem is differ from #241.

And I reported `Content-Type: application/x-ndjson` issue in https://github.com/elastic/elasticsearch-ruby/issues/512.
